File: backend/src/main/java/com/xxgwy/modules/dealer/repository/DealerRepository.java
package com.xxgwy.modules.dealer.repository;

import com.xxgwy.modules.dealer.entity.Dealer;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.UUID;

@Repository
public interface DealerRepository extends JpaRepository<Dealer, UUID> {
}

File: backend/src/main/kotlin/com/xxgwy/modules/dealer/repository/DealerRepository.kt
package com.xxgwy.modules.dealer.repository

import com.xxgwy.modules.dealer.entity.Dealer
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface DealerRepository : JpaRepository<Dealer, UUID> {
    fun findByUserId(userId: UUID): Optional<Dealer>
    fun existsByUserId(userId: UUID): Boolean
}
